src/nodes/node.ts

```diff
@@ -8,11 +8,21 @@ import HTMLElement from './html';
 export default abstract class Node {
 	abstract nodeType: NodeType;
 	public childNodes = [] as Node[];
+	public range: readonly [ number, number ];
 	abstract text: string;
 	abstract rawText: string;
 	// abstract get rawText(): string;
 	abstract toString(): string;
-	public constructor(public parentNode = null as HTMLElement | null) {
+	public constructor(
+		public parentNode = null as HTMLElement | null,
+		range?: [ number, number ]
+	) {
+		Object.defineProperty(this, 'range', {
+			enumerable: false,
+			writable: true,
+			configurable: true,
+			value: range ?? [ -1, -1 ]
+		});
```

Comment:
We define as non-enumerable, to preserve existing functionality for equality tests between a synthetic node (one created manually) and a parsed node (one created during `parse()`). It's unlikely that this has a broad impact scope outside of this library's own tests, but it seems a good idea to make it non-enumerable anyway. It also allows us to not have to change any of our existing tests to exclude `range` property from equality testing.